src/npc_class.cpp

```diff
@@ -244,23 +244,22 @@ void npc_class::load( JsonObject &jo, const std::string & )
         }
     }
 
-    /* Mutation rounds can be specified as either a single distribution:
-     *   "mutation_rounds" : { "constant" : 1 }
-     * or a map of categories to distributions:
-     *   "mutagen_rounds": [
-     *     [ "MUTCAT_ANY", { "constant": 1 } ],
-     *     [ "MUTCAT_INSECT", { "rng": [1, 3] } ]
-     *   ]
+    /* Mutation rounds can be specified as follows:
+     *   "mutation_rounds": {
+     *     "ANY" : { "constant": 1 },
+     *     "INSECT" : { "rng": [1, 3] }
+     *   }
      */
     if( jo.has_object( "mutation_rounds" ) ) {
-        JsonObject jo2 = jo.get_object( "mutation_rounds" );
-        mutation_rounds["MUTCAT_ANY"] = load_distribution( jo2 );
-    } else if( jo.has_array( "mutation_rounds" ) ) {
-        JsonArray jarr = jo.get_array( "mutation_rounds" );
-        while( jarr.has_more() ) {
-            JsonArray jarr2 = jarr.next_array();
-            JsonObject jo2 = jarr2.get_object( 1 );
-            mutation_rounds[jarr2.get_string( 0 )] = load_distribution( jo2 );
+        auto jo2 = jo.get_object( "mutation_rounds" );
+        for( auto &mutation : jo2.get_member_names() ) {
+            auto mutcat = "MUTCAT_" + mutation;
+            if( mutation_category_is_valid( mutcat ) ) {
+                debugmsg( "Unrecognized mutation category %s (i.e. %s)", mutation, mutcat );
```

Comment:
If it *is* valid, you show a debug message? Is there a `!` missing in the condition?

Reply:
Oh gosh, that's embarrassing. I had just noticed and fixed that but forgot to commit before I pushed… fixed now!